spd/utils/module_utils.py

```diff
@@ -1,13 +1,29 @@
+from __future__ import annotations
+
 import fnmatch
 import math
-from typing import Literal
+from dataclasses import dataclass
+from typing import Any, Literal
 
 import torch
 import torch.nn as nn
 from simple_stories_train.models.gpt2_simple import LayerNorm as SSLayerNorm
 from torch import Tensor
 from torch.nn.init import calculate_gain
 
+
+@dataclass
+class ModulePathInfo:
+    """Expanded module path with its number of components.
+
+    Created by expanding ModulePatternInfoConfig patterns against actual module names
+    in the target model. Used internally after pattern expansion.
```

Comment:
I don't think you need that full explanation. This is a pretty simple object.
```suggestion
    """Path to a module (e.g. "h.1.attn.k_proj") and its associated number of components."""
```
Or you can put the example as a comment after `module_path: str`.